src/metabase/db/setup.clj

```diff
@@ -120,6 +124,36 @@
       (log/infof "Successfully verified %s %s application database connection. %s"
                  (.getDatabaseProductName metadata) (.getDatabaseProductVersion metadata) (u/emoji "✅")))))
 
+(mu/defn- check-encryption
+  "Ensure encryption env variable is correctly set if needed, and encrypt the database if it needs to be
+  Encryption status is tracked by an 'encryption-check' value in the settings table.
+  NOTE: the encryption-check setting is not managed like most settings with 'defsetting' so we can manage checking the raw values in the database"
+  [db-type :- :keyword
+   data-source :- (ms/InstanceOfClass javax.sql.DataSource)]
+  (when (or
+         (liquibase/table-exists? "setting" (.getConnection ^javax.sql.DataSource data-source))
+         (liquibase/table-exists? "SETTING" (.getConnection ^javax.sql.DataSource data-source)))
```

Comment:
Is this check needed if this function is called after `run-schema-migrations!`? Presumably the table should exist by this point, and if it doesn't it would be an error state (i.e. `t2/select-one` should fail)

Reply:
I had to add this in for a test that was checking what happens with invalid/incomplete database migrations. It was able to get the database into a state where it got past the run-schema-migrations but there was no tables there. 

I didn't super-like adding this check since it seemed like the run-schema-migrations should make sure it was good and it felt a bit like a test-work-around, but if the test made it through to here it seemed like maybe there were real paths that would also get to that state and I should handle it?

Reply:
I think the DB should be set up by this point, and if it isn't, it's in an invalid state and we should expect the t2 calls to error. So that's why I don't think it would hurt to remove these checks if there's any other way to get the test to pass.

But if not, maybe just add a comment pointing to the test and explaining why it's needed?